code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -269,8 +269,11 @@
 	AddElement(/datum/element/hostile_machine)
 
 /obj/vehicle/sealed/mecha/Destroy()
-	for(var/ejectee in occupants)
-		mob_exit(ejectee, silent = TRUE)
+	/// If the former occupants get polymorphed, mutated, chestburstered,
+	/// or otherwise replaced by another mob, that mob is no longer in .occupants
+	/// and gets deleted with the mech. However, they do remain in .contents
+	for(var/mob/buggy_ejectee in contents)
+		mob_exit(buggy_ejectee, silent = TRUE)
 
```

Comment:
wouldn't the following work and save us the trouble of having to make a whole new list? @san7890

Reply:
i feel weird totally eschewing `occupants` like this because it might open up some really weird edge cases we were previously accounting for so i don't really know

Reply:
you got it; went ahead with your suggested XOR operator